docs/SPEC.md

```diff
@@ -1364,6 +1370,83 @@ A query specification defines what data and operations to perform.
 The execution model reserves the right to perform those operations as efficiently as possible.
 The execution model may rewrite the query in anyway it sees fit while maintaining correctness.
 
+## Versions and editions
+
+Flux follows a [semantic versioning scheme](https://semver.org/) such that breaking changes are clearly communicated as part of the version information.
+*Editions* allow for the introduction of breaking changes via an opt-in mechanism.
+
+Flux editions are a set of features that are enabled in Flux.
+If the edition is not enabled then the features are not enabled.
+These features would otherwise be breaking changes to Flux.
+An edition is explicitly opt-in.
+The pattern of editions allows users to migrate to new Flux versions without risk that their scripts will break.
+
+An edition is separate from a version.
+A version of Flux represents a single point in the commit history of the Flux source code.
+A user can only use a single version of Flux for a given script.
+Editions represent a set of features that are enabled for a given Flux version.
+So long as the Flux version supports all the features of the edition it can be enabled.
+A user can upgrade to a newer Flux version without being required to upgrade to the newest edition of Flux.
+
+New Flux features that require a breaking change to the syntax or semantics of Flux will always be part of a new edition.
+This means that a script can regularly update to the newest Flux version without risk of breaking because any breaking changes are explicitly opt-in.
+With editions the Flux community gets both a pattern where users can always be running the latest version of Flux and the ability to introduce new useful but otherwise breaking changes to Flux.
```

Comment:
This line has me idly thinking about the maintenance impact of supporting multiple editions within the stdlib.

Will we need to manage multiple versions of stdlib, per edition or just on a package by package basis?

Reply:
We will need to manage multiple versions of the stdlib for each edition. This is in the case that the stdlib API changes in an edition (i.e. label polymorphism).